### DISCUSSION TITLE
Possible to run netbox 4.x.. on a ubuntu 18?

### DISCUSSION THREAD

**Discussion by snorte** [netbox-community/netbox#16162] [Help Wanted!]
Is it possible to run and update netbox 3.7.5 running on a ubuntu 18 server to netbox 4?
Got some errors when I tried and roled back.

**Comment by candlerb** [comment]
> Is it possible to run and update netbox 3.7.5 running on a ubuntu 18 server to netbox 4?

With software, anything is possible if you commit sufficient time and effort.

But I don't think anybody here is going to help you do what you're asking for. Ubuntu 18 has gone end-of-life (unless you're paying for extended maintenance?) and comes with ancient versions of dependencies, especially python and postgres.  The time you spend working out how to get sufficiently new versions of those would be *far* better spent simply building an Ubuntu 24.04 virtual machine and installing Netbox 4.

Then all you need to do is [dump the database across](https://docs.netbox.dev/en/stable/administration/replicating-netbox/) and run `./upgrade.sh` on the new server to update the schema (and also copy any custom scripts and media files)

This also has the benefit of being a zero-risk upgrade, as it doesn't alter anything on the original Netbox.
